examples/erc1155/lib.rs

```diff
@@ -0,0 +1,687 @@
+                    AccountId::from([0; 32])
+                } else {
+                    self.env().caller()
+                },
+                token_id: self.token_id_nonce,
+                value,
+            });
+
+            self.token_id_nonce
+        }
+
+        /// Mint a `value` amount of `token_id` tokens.
+        ///
+        /// It is assumed that the token has already been `create`-ed. The newly minted supply will
+        /// be assigned to the caller (a.k.a the minter).
+        ///
+        /// Note that as implemented anyone can mint tokens. If you were to deploy this to a
+        /// production environment you'd probably want to lock down the addresses that are allowed
+        /// to mint tokens.
+        #[ink(message)]
+        pub fn mint(&mut self, token_id: TokenId, value: Balance, _data: Vec<u8>) {
+            assert!(
+                token_id <= self.token_id_nonce,
+                "The `token_id` {:?} has not yet been created in this contract.",
+                token_id
+            );
+
+            self.balances.insert((self.env().caller(), token_id), value);
+
+            // Emit transfer event but with mint semantics
+            self.env().emit_event(TransferSingle {
+                operator: self.env().caller(),
+                from: AccountId::from([0; 32]),
+                to: self.env().caller(),
+                token_id,
+                value,
+            });
+        }
+
+        // Helper function for performing single token transfers.
+        //
+        // Should not be used directly since it's missing certain checks which are important to the
+        // ERC-1155 standard (it is expected that the caller has already perfomred these).
+        fn perform_transfer(
+            &mut self,
+            from: AccountId,
+            to: AccountId,
+            token_id: TokenId,
+            value: Balance,
+            data: Vec<u8>,
+        ) {
+            assert!(
+                self.balance_of(from, token_id) >= value,
+                "Insufficent token balance for transfer."
+            );
+
+            if let Some(b) = self.balances.get_mut(&(from, token_id)) {
+                *b -= value
+            }
+
+            self.balances
+                .entry((to, token_id))
+                .and_modify(|b| *b += value)
+                .or_insert(value);
+
+            self.env().emit_event(TransferSingle {
+                operator: self.env().caller(),
+                from,
+                to,
+                token_id,
+                value,
+            });
+
+            // Quick Haxx, otherwise my tests just panic due to the use of eval_contract()
+            #[cfg(not(test))]
+            {
+                // If our recipient is a smart contract we need to see if they accept or
+                // reject this transfer. If they reject it we need to revert the call.
+                let params = build_call::<ink_env::DefaultEnvironment>()
```

Comment:
It should be possible to just use `Environment` here as an alias for the used environment.

Reply:
Is `Environment` just a type alias for `ink_env::DefaultEnvironment`?

Reply:
in this case, yes. `Environment` within the ink! module is always a type alias for the used environment since every ink! contract so far is concrete over a specific environment. This is because different environments have different properties and therefore the same code could potentially not compile for different environments.

Reply:
With the planned ink! library contracts a contract will be generic over a given contract (and potentially other stuff).

Reply:
Yep, I forgot about this comment